app/controllers/hq/sponsorships_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Hq::SponsorshipsController < HqController
+
+  def inactivate
+    sponsorship = Sponsorship.find(params[:id])
+
+    @sponsorship_inactivator = InactivateSponsorship.new(sponsorship: sponsorship,
+                                                         end_date: params[:sponsorship][:end_date])
+    inactivate_sponsorship and redirect_to hq_sponsor_path(sponsorship.sponsor_id)
+  end
+
+  def destroy
+    sponsorship = Sponsorship.find(params[:id])
+
+    @sponsorship_destructor = DestroySponsorship.new(sponsorship)
+
+    destroy_sponsorship and redirect_to hq_sponsor_path(sponsorship.sponsor_id)
+  end
```

Comment:
notice the sponsor is not loaded anymore
this:
 ```
 sponsor = Sponsor.find(params[:sponsor_id])
 sponsorship = sponsor.sponsorships.find(params[:id])
```
was replaced with
`sponsorship = Sponsorship.find(params[:id])`

and redirecting to `sponsorship.sponsor_id`